### ISSUE TITLE
Turn two is skipped (?)

### ISSUE THREAD

**Issue by dEgolf** [robocode-dev/tank-royale#8]
**Describe the bug**
It seems like the second turn of my robot is skipped.

**To Reproduce**
1. Create a testing robot that prints out the current turn in its Run member. Working in F#, I used the following code:
```
    override this.Run() =  
            while(this.IsRunning) do
                use writer = File.AppendText("log.txt") 
                writer.WriteLine("It is turn: " + (string this.TurnNumber))                 
                this.Go()
```
2. Create a battle with this testing robot and some other robot (I used Target).
3. Let the battle run for a few seconds.
4. Examine the log file "log.txt"

**Expected behavior**
I would expect each robot to be able to act on each turn. In particular, I would expect the log file created above to include the line "It is turn: 2".

**Screenshots**
The start of the resulting log file:
![turn two is skipped](https://user-images.githubusercontent.com/4022661/159094536-ba6d7350-bcf4-43bf-9470-9e3e0b0ab228.png)

**Desktop (please complete the following information):**
 Windows 10

**Java info:**
openjdk version "11.0.14.1" 2022-02-08
OpenJDK Runtime Environment Temurin-11.0.14.1+1 (build 11.0.14.1+1)
OpenJDK 64-Bit Server VM Temurin-11.0.14.1+1 (build 11.0.14.1+1, mixed mode)

**Additional context**
I'm using v0.10.0 of robocode tank-royale.

**Comment by flemming-n-larsen** [comment]
I will have a look into this later.
When creating the sample bots, I saw this behavior once in a while. To me, it looks like some network latency in the beginning when the bot(s) are starting up. Perhaps the WebSocket client needs to "warm-up" to avoid this issue, and the server might give the bots like 10x more time the first 5-10 turns to get up to speed.

Anyways. Since Robocode Tank Royale (RTR) is network-based (web-socket), there is no 100% guarantee that a bot intent will be sent from a bot and received by the server within the time limit for all turns. (I assume that all packages will eventually be sent and received on WebSocket since it is TCP-based).
I propose that a bot should always expect that a turn might be skipped once in a while (for one or the other reason), and make a more robust robot design that is able to cope with that fact.

**Comment by flemming-n-larsen** [comment]
Hi @dEgolf ,
I have been able to reproduce this issue with both the Java and C#/.Net bots. The main bug was a raise condition at the server side, which is now fixed with release [0.13.3].

Note that your bot might skip turns. So you can add a [OnSkippedTurn](https://robocode-dev.github.io/tank-royale/api/dotnet/api/Robocode.TankRoyale.BotApi.BaseBot.html#Robocode_TankRoyale_BotApi_BaseBot_OnSkippedTurn_Robocode_TankRoyale_BotApi_Events_SkippedTurnEvent_) event handler and e.g. write to the console like: `Console.WriteLine("Skipped turn :" + skippedTurnEvent.TurnNumber)` assuming that you run your bot from a console to debug it. 🙂
